### PR TITLE
Add FFmpeg 7.0 support

### PR DESCRIPTION
**Pull request by tszumski**
This PR includes the commits below:
1. Add FFmpeg 7.0 support
2. Use FFmpeg 7.0 as default and update README.md how to switch to FFmpeg 6.1

### CODE REVIEW

File: ffmpeg-plugin/mcm_common.h
Comment (ko80) [inline ffmpeg-plugin/mcm_common.h:19]:
```suggestion
#else /* LIBAVDEVICE_VERSION_MAJOR <= 60 */
```

File: ffmpeg-plugin/mcm_common.h
Comment (ko80) [inline ffmpeg-plugin/mcm_common.h:21]:
```suggestion
#endif /* LIBAVDEVICE_VERSION_MAJOR <= 60 */
```

File: ffmpeg-plugin/mcm_audio_rx.c
Comment (ko80) [inline ffmpeg-plugin/mcm_audio_rx.c:17]:
```suggestion
#endif /* MCM_FFMPEG_7_0 */
```

File: ffmpeg-plugin/mcm_video_rx.c
Comment (ko80) [inline ffmpeg-plugin/mcm_video_rx.c:18]:
```suggestion
#endif /* MCM_FFMPEG_7_0 */
```

**Review by ko80 — CHANGES_REQUESTED** [review]
Please improve README.md to help the user understand how to build for 6.1, and for 7.0.

**Comment by ko80** [comment]
Please update the PR description to provide more details. Example:
```
This PR includes the commits below
* Add ffmpeg 7.0 support
* ...

```

**Review by ko80 — DISMISSED** [review]
Looks good to me

File: ffmpeg-plugin/README.md
Comment (ko80) [inline ffmpeg-plugin/README.md:11]:
```suggestion
1. Clone the FFmpeg repository (Release 7.0 by default) and apply MCM patches
```

File: ffmpeg-plugin/README.md
Comment (ko80) [inline ffmpeg-plugin/README.md:16]:
```suggestion
**Note:** For FFmpeg Release 6.1, replace `7.0` with `6.1` in the above cloning script
```

**Review by ko80 — APPROVED** [review]
Looks good

**Comment by ko80** [comment]
WARNING: Do not merge until Mateusz approves after validating that

**Comment by MateuszGrabuszynski** [comment]
Builds properly. Fallback to 6.1 works as well, but it may be better to mention using a variable instead: `FFMPEG_VER=6.1 ./clone-and-patch-ffmpeg.sh`.

An older version of `libmcm` is expected by the program:
```shell
$ pwd
/home/gta/MCM182

$ git log | head
commit a02720881d5f0fabce635ee5296f2ee057f2cbdf
Author: Tomasz Szumski <tomasz.szumski@intel.com>
Date:   Fri Aug 30 14:13:53 2024 +0200

    Use FFmpeg 7.0 as default and update readme.md how to switch to FFmpeg 6.1
```

```shell
$ LD_LIBRARY_PATH=$(pwd)/ffmpeg-plugin/build/FFmpeg/libavdevice/:$(pwd)/ffmpeg-plugin/build/FFmpeg/libavcodec:$(pwd)/ffmpeg-plugin/build/FFmpeg/libavdevice:$(pwd)/ffmpeg-plugin/build/FFmpeg/libavfilter:$(pwd)/ffmpeg-plugin/build/FFmpeg/libavformat:$(pwd)/ffmpeg-plugin/build/FFmpeg/libavutil:$(pwd)/ffmpeg-plugin/build/FFmpeg/libpostproc:$(pwd)/ffmpeg-plugin/build/FFmpeg/libswresample:$(pwd)/ffmpeg-plugin/build/FFmpeg/libswscale:$(pwd)/out/lib/:$(pwd)/ffmpeg-plugin/build/jpegxs/Bin/Release/ \
ffmpeg -version

ffmpeg: error while loading shared libraries: libmcm_dp.so.24.06.01-31-g8828f22: cannot open shared object file: No such file or directory
```

```shell
$ find . -name libmcm_dp.so.*
./out/lib/libmcm_dp.so.24.06.01-46-ga027208
```

Workaround: Using a library built from commit 6123443 (last `LD_LIBRARY_PATH` argument points to the lib)
```shell
$ LD_LIBRARY_PATH=$(pwd)/ffmpeg-plugin/build/FFmpeg/libavdevice/:$(pwd)/ffmpeg-plugin/build/FFmpeg/libavcodec:$(pwd)/ffmpeg-plugin/build/FFmpeg/libavdevice:$(pwd)/ffmpeg-plugin/build/FFmpeg/libavfilter:$(pwd)/ffmpeg-plugin/build/FFmpeg/libavformat:$(pwd)/ffmpeg-plugin/build/FFmpeg/libavutil:$(pwd)/ffmpeg-plugin/build/FFmpeg/libpostproc:$(pwd)/ffmpeg-plugin/build/FFmpeg/libswresample:$(pwd)/ffmpeg-plugin/build/FFmpeg/libswscale:$(pwd)/out/lib/:$(pwd)/ffmpeg-plugin/build/jpegxs/Bin/Release/:$(pwd)/../MCM/out/lib \
ffmpeg -version
ffmpeg version 5148262 Copyright (c) 2000-2024 the FFmpeg developers
built with gcc 11 (Ubuntu 11.4.0-1ubuntu1~22.04)
configuration: --enable-shared --enable-mcm
libavutil      59.  8.100 / 59.  8.100
libavcodec     61.  3.100 / 61.  3.100
libavformat    61.  1.100 / 61.  1.100
libavdevice    61.  1.100 / 61.  1.100
libavfilter    10.  1.100 / 10.  1.100
libswscale      8.  1.100 /  8.  1.100
libswresample   5.  1.100 /  5.  1.100
```

Working on further tests.

**Comment by MateuszGrabuszynski** [comment]
ST20 video test went fine with sample bars video. Planning to test at least also with ST30.
```shell
$ md5sum out_auto_2024-09-02_074852.yuv tests/single-node-sample-apps/video.yuv 
09e2ab3d78d3762993a48824e7ffb693  out_auto_2024-09-02_074852.yuv
09e2ab3d78d3762993a48824e7ffb693  tests/single-node-sample-apps/video.yuv
```

**Review by MateuszGrabuszynski — APPROVED** [review]
Looks good! 🚀
PS: The issue with lib version was due to not following the instruction (prerequisites were installed after the plugin build).

File: ffmpeg-plugin/clone-and-patch-ffmpeg.sh
Comment (Mionsz) [inline ffmpeg-plugin/clone-and-patch-ffmpeg.sh:13]: Please change this to latest patch version as there are lots of fixes there.

```suggestion
# Default to latest 7.0.2
FFMPEG_VER="${FFMPEG_VER:-7.0.2}"
```

File: ffmpeg-plugin/7.0/0001-Enable-FFmpeg-support-in-MCM.patch
Comment (Mionsz) [inline ffmpeg-plugin/7.0/0001-Enable-FFmpeg-support-in-MCM.patch:1]: Probably this file will need to be adjusted also before merge.

**Review by Mionsz — CHANGES_REQUESTED** [review]

File: ffmpeg-plugin/clone-and-patch-ffmpeg.sh
Comment (ko80) [reply ffmpeg-plugin/clone-and-patch-ffmpeg.sh:13]: I don't get it. We're cloning FFmpeg from the release branch, not the tag
```bash
# Clone FFmpeg:
git clone --branch "release/${FFMPEG_VER}" --depth 1 https://github.com/FFmpeg/FFmpeg.git "${BUILD_DIR}/FFmpeg"
```
This means if you clone it today, it will include all commits related to `7.0.2` issued recently.

**Comment by Mionsz** [comment]
@Mionsz @tszumski as you showed the z (patch) from semver x.y.z. is automatically included in the 7.0 branch. So no changes are required.

**Review by Mionsz — APPROVED** [review]

**Pull request merged**
